akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2Demux.scala

```diff
@@ -277,7 +280,7 @@ private[http2] abstract class Http2Demux(http2Settings: Http2CommonSettings, ini
 
         allowReadingIncomingFrames = allow
       }
-      def pullFrameIn(): Unit = if (allowReadingIncomingFrames && !hasBeenPulled(frameIn)) pull(frameIn)
+      def pullFrameIn(): Unit = if (allowReadingIncomingFrames && !hasBeenPulled(frameIn) && !isClosed(frameIn)) pull(frameIn)
```

Comment:
when improving the shutdown sequence I stumbled upon scenarios where we tried to pull from a closed inlet.

Reply:
Good catch

Reply:
This might be because of the `onUpstreamFinish` for `frameIn`. If we can remove that handling, we should also remove this one.

Reply:
This is indeed still needed even without the `frameIn` handling.